client/asset/eth/multirpc.go

```diff
@@ -803,9 +808,21 @@ func (m *multiRPCClient) withOne(providers []*provider, f func(*provider) error,
 		readyProviders = providers
 	}
 	for _, p := range readyProviders {
-		err := f(p)
+		ctx, cancel := context.WithTimeout(ctx, defaultRequestTimeout)
+		// Fetching the best header will check that either the
+		// provider's cached header is not too old or that a
+		// newly fetched header is not too old. If it is too
+		// old that indicates the provider is not in sync and
+		// should not be used.
+		if _, err := p.bestHeader(ctx, m.log); err != nil {
```

Comment:
So in `(*multiRPCClient).bestHeader`, it's going to do: https://github.com/decred/dcrdex/blob/e598ecda67878df8bc302fd6e88ba60803bfed76/client/asset/eth/multirpc.go#L960-L963

which means that through `withAny` -> `withOne`, we're going to get here and do `(*provider).bestHeader` twice in a row?

I kinda wish something other than `withOne` were in charge of flagging providers as stale.  `(*ETHWallet).checkForNewBlocks` is going to be calling `bestHeader` regularly anyway, so it seems like either `(*provider) bestHeader` or `(*multiRPCClient) bestHeader`would be a good place to set some flag, which `withOne` could then use in a similar manner to how it gets `readyProviders` with `!p.failed()`.

Reply:
Can I stick the check in a goroutine? I'm afraid if you just put the check in a random call, it randomly might not get called while another random call is called and the stale provider is used.

Reply:
While adding the goroutine, it seems like the other goroutine `subscribeHeaders` will continue to run on old providers if you reconfigure, although it does not do anything, it's just waiting. Maybe should be stopped when the rpc client is closed.

Reply:
Is this ok? Just check every 50 seconds. https://github.com/decred/dcrdex/compare/e598ecda67878df8bc302fd6e88ba60803bfed76..f981b0e14dea6c63eee50728992d19cf321694cd

Reply:
Oh, maybe I was mistaken.

Reply:
It makes sense to wait for it to finish during shutdown?

Reply:
I tested some more and you are right, closing the client causes the 'ok' part of `case err, ok := <-sub.Err():` being true.

I still think the extra monitoring is sane. Can revert though.